src/pid/environ.rs

```diff
@@ -0,0 +1,78 @@
+//! Process initial environment from `/proc/[pid]/environ`.
+
+use std::ffi::{OsStr, OsString};
+use std::fs::File;
+use std::io::{Result, Read};
+use std::os::unix::ffi::OsStrExt;
+use std::path::Path;
+
+use parsers::map_result;
+use nom::IResult;
+
+use libc::pid_t;
+
+/// A list of environment variables and their values.
+pub type Environ = Vec<(OsString, OsString)>;
```

Comment:
I was hoping there was a way to avoid collecting into an intermediate `Vec`, but I spent a while on it and didn't find an elegant way.  Since the API's just exposing a vec, I think we should skip the type def and just return `Vec<(OsString, OsString)>`.

Reply:
Hm. I'll think a bit more on this.